AlexaSkillsKit.Lib/Helpers/DateTimeUtils.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Globalization;
+
+namespace AlexaSkillsKit.Helpers
+{
+    public class DateTimeUtils
+    {
+        public static DateTime FromAmazonRequest(string datetimeString)
+        {
+            try
+            {
+                DateTime datetime;
+
+                //Try a generic parse
+                if (DateTime.TryParse(datetimeString, CultureInfo.InvariantCulture, DateTimeStyles.None, out datetime))
```

Comment:
For consistency with the rest of the library it would be better to use `Json.Net` to do the date/time parsing rather than `DateTime.Parse` directly (also so it maintains the same behavior as prior to the fix). Case in point, the input `2017/7.14` which you consider bad input in your unit tests is actually valid according to `Json.Net`. You might disagree but I think consistency across the entire library is more important.